test/builder-api/src/jobs.js

```diff
@@ -53,7 +53,7 @@ describe('Jobs API', function () {
         .type('application/json')
         .accept('application/json')
         .set('Authorization', global.boboBearer)
-        .expect(400)
+        .expect(404)
```

Comment:
This change is to work around the difference in default configuration between builder-api and jobsrv.  In builder-api, kernel2 is  part of the default build groups, but in jobsrv it is not. This causes an error to bubble back from jobsrv as a 404, rather than a 400.

```[builder-api.default][2019-09-19T19:03:42Z INFO  actix_web::middleware::logger] 127.0.0.1:34348 "POST /v1/depot/pkgs/schedule/neurosis/testapp HTTP/1.1" 403 0 "-" "node-superagent/3.8.1" 0.002317[0m
[builder-api.default][2019-09-19T19:03:42Z DEBUG habitat_builder_api::server::framework::middleware] HAB_FUNC_TEST: Some("1"); calling session_create_short_circuit
[builder-api.default][2019-09-19T19:03:42Z DEBUG habitat_builder_api::server::framework::middleware] issuing session, id: 1339271693520084992 email: "bobo@example.com" name: "bobo" token: "CIDAgPCmkIPLEhIBMBgAIgRib2Jv" flags: 0 oauth_token: "bobo"
[builder-api.default][2019-09-19T19:03:42Z DEBUG builder_core::rpc] Sending RPC Message: JobGroupSpec
[builder-jobsrv.default[2019-09-19T19:03:42Z DEBUG habitat_builder_jobsrv::server] Got RPC message, body =
[builder-jobsrv.default]Json: RpcMessage { id: "JobGroupSpec", body: [10, 8, 110, 101, 117, 114, 111, 115, 105, 115, 18, 7, 116, 101, 115, 116, 97, 112, 112, 24, 0, 34, 20, 120, 56, 54, 95, 54, 52, 45, 108, 105, 110, 117, 120, 45, 107, 101, 114, 110, 101, 108, 50, 40, 0, 48, 0, 56, 0, 64, 128, 192, 128, 240, 166, 144, 131, 203, 18, 74, 4, 98, 111, 98, 111] }
[builder-jobsrv.default][2019-09-19T19:03:42Z DEBUG habitat_builder_jobsrv::server::handlers] job_group_create message: origin: "neurosis" package: "testapp" deps_only: false target: "x86_64-linux-kernel2" origin_only: false package_only: false trigger: Unknown requester_id: 1339271693520084992 requester_name: "bobo"
[builder-jobsrv.default][2019-09-19T19:03:42Z DEBUG habitat_builder_jobsrv::server::handlers] Rejecting build request with target: PackageTarget(X86_64_Linux_Kernel2)
[builder-jobsrv.default][2019-09-19T19:03:42Z INFO  actix_web::middleware::logger] 172.17.0.2:50548 "POST /rpc HTTP/1.1" 404 0 "-" "Habitat-Builder" 0.000935
[builder-api.default][2019-09-19T19:03:42Z DEBUG reqwest::async_impl::response] Response: '404 Not Found' for http://172.17.0.2:5580/rpc

Reply:
Ah, we might as well make the jobsrv build_targets include kernel2 so we are consistent with the api

Reply:
I've added that to the default, and removed the `does not build kernel2` test as that no longer makes sense.